pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -543,6 +543,17 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 	return out, nil
 }
 
+// GetProxyLocality returns the locality where the proxy runs.
+func (c *Controller) GetProxyLocality(proxy *model.Proxy) string {
```

Comment:
doesn't the serviceInstance or networkEndpoint have a locality field? Why not populate it as part of the GetProxyServiceInstances?

Reply:
Well, `GetProxyServiceInstances` is a far more heavy call.